applications/nrf5340_audio/src/amod_main.c

```diff
@@ -0,0 +1,321 @@
+		return -1;
+	}
+
+	ret = amod_start(handle);
+	if (ret) {
+		printf("Start FAILED test, ret %d", ret);
+		return -1;
+	}
+
+	state = amod_state_get(handle);
+	if (state != AMOD_STATE_RUNNING) {
+		printf("Failed to get the correct state for module %s instance %s, ret %d",
+		       base_name, instance_name, ret);
+		return -1;
+	}
+
+	ret = amod_pause(handle);
+	if (ret) {
+		printf("Send command FAILED test, ret %d", ret);
+		return -1;
+	}
+
+	state = amod_state_get(handle);
+	if (state != AMOD_STATE_PAUSED) {
+		printf("Failed to get the correct state for module %s instance %s, ret %d",
+		       base_name, instance_name, ret);
+		return -1;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Function to start a single LC3 decoder module and configure it.
+ *
+ */
+static int lc3_decoder_static_test(void)
+{
+	int ret;
+	struct amod_handle handle;
+	struct amod_parameters lc3_dec_parameters;
+	char *in_msg;
+	char *out_msg;
+	char *data;
+
+	amod_parameters_configure(lc3_dec_parameters, lc3_dec_description, lc3_decoder_thread_stack,
+				  LC3_DECODER_THREAD_STACK_SIZE, LC3_DECODER_THREAD_PRIORITY,
+				  LC3_DECODER_IN_MESSAGE_NUM, LC3_DECODER_OUT_MESSAGE_NUM);
+	if (ret) {
+		printf("Thread configuration FAILED , ret %d", ret);
+		return -1;
+	}
+
+	ret = static_allocate_memory(&handle, &in_msg, &out_msg, &data);
+	if (ret) {
+		printf("Failed to allocate memory, ret %d", ret);
+		return -1;
+	}
+
+	ret = basic_test(&lc3_dec_parameters, in_msg, out_msg, data, handle);
+	if (ret) {
+		printf("Failed to basic tests, ret %d", ret);
+		return -1;
+	}
+
+	ret = amod_close(&handle);
+	if (ret) {
+		printf("Close FAILED, ret %d", ret);
+		return -1;
+	}
+
+	if (handle != NULL) {
+		printf("Close failed to NULL handle");
+		return -1;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Function to start a single module to the configured stage.
+ *
+ */
+static int void lc3_decoder_dynamic_test(void)
+{
+	int ret;
+	struct amod_handle handle;
+	struct amod_handle handle_memory;
+	struct amod_parameters lc3_dec_parameters;
+	char *in_msg;
+	char *out_msg;
+	char *data;
+
+	amod_parameters_configure(lc3_dec_parameters, lc3_dec_description, lc3_decoder_thread_stack,
+				  LC3_DECODER_THREAD_STACK_SIZE, LC3_DECODER_THREAD_PRIORITY,
+				  LC3_DECODER_IN_MESSAGE_NUM, LC3_DECODER_OUT_MESSAGE_NUM);
+	if (ret) {
+		printf("Thread configuration FAILED , ret %d", ret);
+		return -1;
+	}
+
+	ret = dynamic_allocate_memory(lc3_dec_parameters, &lc3_dec_inst_1_config, &handle, &in_msg,
+				      &out_msg, &data);
+	if (ret) {
+		printf("Failed to allocate memory, ret %d", ret);
+		return -1;
+	}
+
+	ret = basic_test(in_msg, out_msg, data, handle);
+	if (ret) {
+		printf("Failed to basic tests, ret %d", ret);
+		return -1;
+	}
+
+	handle_memory = handle;
+
+	ret = amod_close(&handle);
+	if (ret) {
+		printf("Close FAILED, ret %d", ret);
+		return -1;
+	}
+
+	if (handle != NULL) {
+		printf("Close failed to NULL handle");
+		return -1;
+	}
+
+	k_free(in_msg);
```

Comment:
Discuss: Should we have a helper function for this?

Reply:
It will be dependent on prefered memory management system. If we do I would suggest a mem_mngr API for clarity.